test/tests/mail/model/MailListModelTest.ts

```diff
@@ -0,0 +1,481 @@
+import o from "../../../../packages/otest/dist/otest"
+import { LoadedMail, MailListModel } from "../../../../src/mail-app/mail/model/MailListModel"
+import {
+	createMailFolder,
+	createMailSetEntry,
+	Mail,
+	MailboxGroupRootTypeRef,
+	MailBoxTypeRef,
+	MailFolder,
+	MailFolderTypeRef,
+	MailSetEntry,
+	MailSetEntryTypeRef,
+	MailTypeRef,
+} from "../../../../src/common/api/entities/tutanota/TypeRefs"
+import { matchers, object, verify, when } from "testdouble"
+import { ConversationPrefProvider } from "../../../../src/mail-app/mail/view/ConversationViewModel"
+import { EntityClient } from "../../../../src/common/api/common/EntityClient"
+import { MailModel } from "../../../../src/mail-app/mail/model/MailModel"
+import { InboxRuleHandler } from "../../../../src/mail-app/mail/model/InboxRuleHandler"
+import { ExposedCacheStorage } from "../../../../src/common/api/worker/rest/DefaultEntityRestCache"
+import { MailSetKind, OperationType } from "../../../../src/common/api/common/TutanotaConstants"
+import {
+	constructMailSetEntryId,
+	CUSTOM_MAX_ID,
+	deconstructMailSetEntryId,
+	elementIdPart,
+	GENERATED_MAX_ID,
+	getElementId,
+	getListId,
+	isSameId,
+} from "../../../../src/common/api/common/utils/EntityUtils"
+import { PageSize } from "../../../../src/common/gui/base/ListUtils"
+import { createTestEntity } from "../../TestUtils"
+import { tutaDunkel, tutaRed } from "../../../../src/common/gui/builtinThemes"
+import { EntityUpdateData } from "../../../../src/common/api/common/utils/EntityUpdateUtils"
+import { MailboxDetail } from "../../../../src/common/mailFunctionality/MailboxModel"
+import { GroupInfoTypeRef, GroupTypeRef } from "../../../../src/common/api/entities/sys/TypeRefs"
+import { ConnectionError } from "../../../../src/common/api/common/error/RestError"
+import { clamp, pad } from "@tutao/tutanota-utils"
+
+o.spec("MailListModelTest", () => {
+	let model: MailListModel
+
+	const mailboxDetail: MailboxDetail = {
+		mailbox: createTestEntity(MailBoxTypeRef),
+		mailGroupInfo: createTestEntity(GroupInfoTypeRef),
+		mailGroup: createTestEntity(GroupTypeRef),
+		mailboxGroupRoot: createTestEntity(MailboxGroupRootTypeRef),
+	}
+
+	const mailSetEntriesListId = "entries"
+	const _ownerGroup = "me"
+
+	const labels: MailFolder[] = [
+		createTestEntity(MailFolderTypeRef, {
```

Comment:
just fyi: you don't need to specify all the fields if you use `createTestEntity()`